### ISSUE TITLE
Pane doesn't resize correctly after pane deletion

### ISSUE THREAD

**Issue by amir20** [antoniandre/splitpanes#65]
Hello,

I am not sure if I have discovered a bug. But when I go from two panes to one, then the width of the remaining pane is still 50%. 

My code looks like this: 

```html
<splitpanes class="default-theme">
        <pane>
          <router-view></router-view>
        </pane>
        <pane v-for="other in activeContainers" :key="other.id">
          <div class="name columns is-marginless">
            <span class="column">{{ other.name }}</span>
            <span class="column is-narrow">
              <button class="delete is-medium" @click="removeActiveContainer(other)"></button>
            </span>
          </div>
          <log-event-source :id="other.id" v-slot="eventSource">
            <log-viewer :messages="eventSource.messages"></log-viewer>
          </log-event-source>
        </pane>
      </splitpanes>
```

After adding a new pane it looks like this. 

<img width="1676" alt="Screen Shot 2019-11-15 at 12 00 13 PM" src="https://user-images.githubusercontent.com/260667/68971951-8b827a80-079f-11ea-9c5c-2a24d2174876.png">

However, after removing the last pane, I still see 50 width. 

<img width="1679" alt="Screen Shot 2019-11-15 at 12 00 21 PM" src="https://user-images.githubusercontent.com/260667/68971974-976e3c80-079f-11ea-8c66-307b6a0e6b88.png">

Is there a way to refresh the size?

**Comment by amir20** [comment]
I went through the code a little and it's unclear how `this.$children` is watched for changes. Let me know @antoniandre how I can help debug.

**Comment by amir20** [comment]
Here is a codepen to reproduce problem https://codepen.io/antoniandre/pen/XybPKP

**Comment by antoniandre** [comment]
Hi @amir20, Thanks for reporting.
Yes this is a new bug, I will look into that soon and will let you know.

**Comment by amir20** [comment]
Thanks @antoniandre. I'd imagine this a major bug. Let me know if I can help.

**Comment by xiaolee55** [comment]
I get this bug ，too

**Comment by antoniandre** [comment]
Hi @amir20, thanks for suggesting help, I started to look into that, the problem at the moment is that after a deletion occurs, the blank space should not be just distributed to all the other panes equally, but we should also check the min and max of each panes before applying new width/height to make sure we don't break the panes constraints.
E.g:
3 panes, 33.33%, 33.33%, 33.33%, first pane has a max width of 40%.
when deleting the last pane, the new panes width should be 40%, 60%, not 50%, 50%.

This is not yet in the codebase, so I need to add those functions.

Will keep you posted.

**Comment by antoniandre** [comment]
Hi @amir20, I published a patched (2.1.1) to fix this.
It will take in account the max size of each the current panes when allocating an equal share of the white space left after a pane deletion.
It will also try to keep panes collapsed if it can, and if the white space is too big to distribute only to expanded panes it will also expand the collapsed panes.

I will add a similar function to check the panes constraints when adding panes soon. At the moment it does not care about the min property of the current panes when adding a new one, which can result in a wrong display.

Caveat:
If any of the calculation above fails, it will fail gracefully in spite of the visual mismatch and won't say anything: I don't think I should add any console log/warn in a library, this could fail on older IE and should not be there in production servers.

Let me know if you have a feedback on that.

Cheers.

**Comment by amir20** [comment]
Thanks @antoniandre I'll try it this week.
